File: src/interface/dropdowns/properties.py
from PyQt6.QtCore import Qt
from PyQt6.QtWidgets import QCheckBox, QVBoxLayout, QWidget


class StrataDropdownProperties(QWidget):
    """
    Shows the different list columns to enable/disable
    """

    def __init__(self) -> None:
        """
        Constructor
        """

        super().__init__()
        self.setWindowFlags(self.windowFlags() | Qt.WindowType.Popup)

        self.check_name = QCheckBox(text="file name")
        self.check_path = QCheckBox(text="file path")
        self.check_size = QCheckBox(text="file size")
        self.check_extension = QCheckBox(text="file extension")
        self.check_modified = QCheckBox(text="date modified")

        # set defaults
        self.check_name.setCheckState(Qt.CheckState.Checked)

        _layout_main = QVBoxLayout(self)
        _layout_main.addWidget(self.check_name)
        _layout_main.addWidget(self.check_path)
        _layout_main.addWidget(self.check_size)
        _layout_main.addWidget(self.check_extension)
        _layout_main.addWidget(self.check_modified)

File: src/dependency_viewer/interface/widgets/dropdown/Properties.cpp
#include "Properties.h"
#include <QVBoxLayout>

PropertiesDropdownWidget::PropertiesDropdownWidget(QWidget *parent) {
    setWindowFlags(Qt::WindowFlags(windowFlags() | Qt::WindowType::Popup));
    
    QVBoxLayout *layout = new QVBoxLayout(this);
    checkName_ = new QCheckBox("File Name");
    checkPath_ = new QCheckBox("File Path");
    checkSize_ = new QCheckBox("File Size");
    checkExtension_ = new QCheckBox("File Extension");
    checkModified_ = new QCheckBox("Date Modified");

    checkName_->setCheckState(Qt::CheckState::Checked);

    layout->addWidget(checkName_);
    layout->addWidget(checkPath_);
    layout->addWidget(checkSize_);
    layout->addWidget(checkExtension_);
    layout->addWidget(checkModified_);
}
